app/modules/report/audio_event_report.rb

```diff
@@ -0,0 +1,259 @@
+# frozen_string_literal: true
+
+module Report
+  # I think these report classes won't be needed as the new appraoch is
+  # progressed? And the logic for a report configuration can move to the
+  # report's controller. So this is inbetween the old and new approach.
+  class AudioEventReport < Base
+    include Report::ArelHelpers
+
+    BASE_TABLE = Arel::Table.new('base_table')
+
+    def build_query
+      base_query_projected = @base_query.arel.project(attributes)
+      base_query_joined = add_joins(base_query_projected)
+      base_table = Arel::Table.new('base_table')
+      base_cte = Arel::Nodes::As.new(base_table, base_query_joined)
+
+      time_series_options = TimeSeries::Options.call(@parameters, base_table: BASE_TABLE)
+      composition_options = {
+        bucketed_time_series: Section::Accumulation::TABLE_BUCKETED_TIME_SERIES,
+        base_table: BASE_TABLE,
+        base_verification: Section::EventSummary::TABLE_VERIFICATION_BASE
+      }
+      recording_coverage_options = Report::Section::Coverage.options(time_series_options) { |opt|
+        opt[:source] = base_table
+        opt[:lower_field] = base_table[:recorded_date]
+        opt[:upper_field] = Report::ArelHelpers.arel_recorded_end_date(base_table) # using AudioRecording.arel_recorded_end_date => missing FROM-clause entry for table "audio_recordings"
```

Comment:
make the function on AudioRecording accept an optional arel_table argument. If not specified just defaults to `self.class.arel_table`